doc/2/api/controllers/document/m-upsert/index.md

```diff
@@ -0,0 +1,173 @@
+---
+code: true
+type: page
+title: mUpsert
+---
+
+# mUpsert
+
+<SinceBadge version="auto-version"/>
+
+Applies partial changes to multiple document. If a document doesn't already exist, a new document is created.
+
+::: info
+The number of documents that can be updated by a single request is limited by the `documentsWriteCount` server configuration (see the [Configuring Kuzzle](/core/2/guides/advanced/configuration) guide).
+:::
+
+---
+
+## Query Syntax
+
+### HTTP
+
+```http
+URL: http://kuzzle:7512/<index>/<collection>/_mUpsert[?refresh=wait_for][&retryOnConflict=<retries>][&silent]
+Method: POST
+Body:
+```
+
+```js
+{
+  "documents": [
+    {
+      "_id": "<documentId>",
+      "changes": {
+        // document partial changes
+      },
+      "default": {
+        // optional: document fields to add to the "update" part if the document
+        // is created
+      }
+    },
+    {
+      "_id": "<anotherDocumentId>",
+      "changes": {
+        // document partial changes
+      },
+    }
+  ]
+}
+```
+
+### Other protocols
+
+```js
+{
+  "index": "<index>",
+  "collection": "<collection>",
+  "controller": "document",
+  "action": "mUpsert",
+  "body": {
+    "documents": [
+      {
+        "_id": "<documentId>",
+        "changes": {
+          // document partial changes
+        },
+        "default": {
+          // optional: document fields to add to the "update" part if the document
+          // is created
+        }
+      },
+      {
+        "_id": "<anotherDocumentId>",
+        "changes": {
+          // document partial changes
+        },
+      }
+    ]
+  }
+}
+```
+
+### Kourou
+
+```bash
+kourou document:mUpsert <index> <collection> <body>
+kourou document:mUpsert <index> <collection> <body> -a silent=true
+```
+
+---
+
+## Arguments
+
+- `collection`: collection name
+- `index`: index name
+
+### Optional:
+
+- `refresh`: if set to `wait_for`, Kuzzle will not respond until the updates are indexed
+- `retryOnConflict`: conflicts may occur if the same document gets updated multiple times within a short timespan in a database cluster. You can set the `retryOnConflict` optional argument (with a retry count), to tell Kuzzle to retry the failing updates the specified amount of times before rejecting the request with an error.
+- `silent`: if set, then Kuzzle will not generate notifications
+
```

Comment:
Can we have a `strict` option as well on this API action?

Reply:
Sure, I missed that because it's in another of my PR and still not in 2-dev!